bentoml/deployment/clipper/__init__.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import shutil
+import re
+
+import docker
+
+from bentoml.archive import load
+from bentoml.handlers import ImageHandler
+from bentoml.deployment.utils import (
+    generate_bentoml_deployment_snapshot_path,
+    process_docker_api_line,
+)
+from bentoml.deployment.clipper.templates import (
+    DEFAULT_CLIPPER_ENTRY,
+    DOCKERFILE_CLIPPER,
+)
+from bentoml.exceptions import BentoMLException
+
+
+def generate_clipper_compatiable_string(item):
+    """Generate clipper compatiable string. It must be a valid DNS-1123.
+    It must consist of lower case alphanumeric characters, '-' or '.',
+    and must start and end with an alphanumeric character
+
+    :param item: String
+    :return: string
+    """
+
+    pattern = re.compile("[^a-zA-Z0-9-]")
+    result = re.sub(pattern, "-", item)
+    return result.lower()
+
+
+def deploy_bentoml(
+    clipper_conn, archive_path, api_name, input_type='strings', labels=["bentoml"]
+):
+    """Deploy bentoml bundle to clipper cluster
+
+    Args:
+        clipper_conn(clipper_admin.ClipperConnection): Clipper conntection instance
+        archive_path(str): Path to the bentoml service archive.
+        api_name(str): name of the api that will be running in the clipper cluster
+        input_type(str): Input type that clipper accept
```

Comment:
add some more notes here - especially about the default value and behavior around ImageHnalder check below

Reply:
done